charts/telemetry-collector/templates/configmap.yaml

```
@@ -0,0 +1,78 @@
+{{- include
+      "accelleran.common.configMap"
+      (fromYaml (include "accelleran.telemetry-collector.configMap.args" .))
+-}}
+
+{{- define "accelleran.telemetry-collector.configMap.args" -}}
+{{- $ := . -}}
+
+top:
+  {{ $ | toYaml | nindent 2 }}
+
+name: {{ include "accelleran.common.fullname" (dict "top" $) }}-config
+
+data:
+  config.yaml: |-
+    internal:
+      kafka:
+        url: "{{ include "accelleran.common.bootstrap.kafka.hostname" (dict "top" .) }}:{{ include "accelleran.common.bootstrap.kafka.port" (dict "top" .) }}"
+      {{- if $.Values.config.influxDB }}
+      influxDB:
+        {{- if $.Values.config.influxDB.url }}
+        url: "{{ $.Values.config.influxDB.url }}"
+        {{ else }}
+        url: "http://{{ $.Release.Name }}-influxdb2 }}"
+        {{- end }}
+        org: "{{ $.Values.config.influxDB.org }}"
+      {{- end }}
+    pipelines:
+      {{- $natsHostname := include "accelleran.common.bootstrap.nats.hostname" (dict "top" .) }}
+      {{- $natsPort := include "accelleran.common.bootstrap.nats.port" (dict "top" .) }}
+      {{- range $pipeline := $.Values.config.pipelines }}
+      - name: "{{ $pipeline.name }}" 
+        component: "{{ $pipeline.component }}"
+        vendor: "{{ $pipeline.vendor }}"
+        enabled: "{{ $pipeline.enabled }}"
+        {{- if $pipeline.consumers }}
+        consumers:
+          {{- if $pipeline.consumers.natsConsumer }}
+            natsConsumer:
+              {{- if $pipeline.consumers.natsConsumer.url }}
+              url: "{{ $pipeline.consumers.natsConsumer.url }}"
+              {{ else }}
+              url: "{{ $natsHostname }}:{{ $natsPort }}"
+              {{- end }}
+              subjects: "{{ $pipeline.consumers.natsConsumer.subjects }}"
+          {{- end }}
+          {{- if $pipeline.consumers.udpConsumer }}
+            udpConsumer:
+              port: "{{ $pipeline.consumers.udpConsumer.port }}"
+          {{- end }}
```

Lots of approaches possible here, but I would maybe opt for something that requires a bit less changes if we were to add config:

```suggestion
          {{- with $pipeline.consumers.udpConsumer }}
            udpConsumer: {{ . | toYaml | nindent 14 }}
          {{- end }}
```

I'm not completely sure if this works out of the box, but at least the idea is here. Two things to note:
  * **n**indent so that the yaml string starts on the next line)
  * `with` instead of `if` so that the variable is stored within `.`

---

Thanks